### PR TITLE
A window that shows mean orbital elements for the active vessel

### PR DESCRIPTION
**Pull request by eggrobin**
Example with 100 days on a Молния orbit:
screenshot28

### CODE REVIEW

File: ksp_plugin_adapter/orbit_analyser.cs [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+          value      : 0,
+          size       : (float)analysis.progress_of_next_analysis,
+          leftValue  : 0,
+          rightValue : 1);
+
+      OrbitalElements? elements = null;
+      OrbitRecurrence? recurrence = null;
+      OrbitGroundTrack? ground_track = null;
+      double mission_duration = analysis.mission_duration;
+      if (analysis.elements_has_value) {
```
Comment (pleroy) [inline ksp_plugin_adapter/orbit_analyser.cs]: Create a bug for supporting optional fields in the auxiliary interface messages.

File: ksp_plugin_adapter/orbit_analyser.cs [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+        equatorial_crossings?.longitudes_reduced_to_ascending_pass.
+            FormatEquatorialAngleInterval(primary));
+    LabeledField(
+        "Descending pass",
+        equatorial_crossings?.longitudes_reduced_to_descending_pass.
+            FormatEquatorialAngleInterval(primary));
+  }
+
+  private void LabeledField(
+      string label,
```
Comment (pleroy) [inline ksp_plugin_adapter/orbit_analyser.cs]: Would fit on the previous line.

File: ksp_plugin_adapter/orbit_analyser.cs [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+    using (new UnityEngine.GUILayout.HorizontalScope()) {
+      UnityEngine.GUILayout.Label(label);
+      UnityEngine.GUILayout.Label(value ?? em_dash,
+          Style.RightAligned(UnityEngine.GUI.skin.label));
+    }
+  }
+
+  private const string em_dash = "—";
+
+  private IntPtr plugin => adapter_.Plugin();
```
Comment (pleroy) [inline ksp_plugin_adapter/orbit_analyser.cs]: Move up.

File: ksp_plugin_adapter/orbit_analyser.cs (hunk 1 above)
Comment (eggrobin) [reply ksp_plugin_adapter/orbit_analyser.cs]: #2339.

**Pull request merged**